File: TrainingTracker/backend/src/TrainingTracker.Domain/Entities/CourseAssignment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using TrainingTracker.Domain.Enums;

namespace TrainingTracker.Domain.Entities;

public class CourseAssignment
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    // Foreign keys
    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; } = null!;

    public Guid CourseId { get; set; }
    public Course Course { get; set; } = null!;

    // Assignment details
    public DateTime AssignedAt { get; set; } = DateTime.UtcNow;
    public DateTime? DueAt { get; set; }
    public DateTime? CompletedAt { get; set; }

    public AssignmentStatus Status { get; set; } = AssignmentStatus.Assigned;

    [MaxLength(1000)]
    public string? Notes { get; set; }
}

File: TrainingTracker/frontend/training-tracker-ui/src/app/shared/models/course-assignment.ts
import { AssignmentStatus } from './assignment-status.enum';

export interface CourseAssignment {
  id: string;
  employeeId: string;
  employeeName: string;
  courseId: string;
  courseTitle: string;
  assignedAt: Date;
  dueAt?: Date;
  completedAt?: Date;
  status: AssignmentStatus;
  notes?: string;
}

export interface CreateAssignment {
  employeeId: string;
  courseId: string;
  dueAt?: Date;
  notes?: string;
}

export interface UpdateAssignmentStatus {
  id: string;
  status: AssignmentStatus;
  completedAt?: Date;
  notes?: string;
}

export { AssignmentStatus };
